### PR TITLE
Rhcloud 19990 integrate with compliance

### PR DESCRIPTION
**Pull request by dagbay-rh**
Added an endpoint that acts as a proxy to export compliance service. The endpoint accepts a request meant for export compliance, forwards it, then returns the raw response. The request/response schema can be found here: [export-compliance-useful-queries](https://gitlab.corp.redhat.com/it-legal/export-compliance-useful-queries/-/tree/main#service-endpoints)

Ticket: RHCLOUD-19990

Original pr: https://github.com/RedHatInsights/entitlements-api-go/pull/88, had to recreate remote branch which included creating a new pr

# Platform Security

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist 

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Comment by coderbydesign** [comment]
@wcmitchell @mjholder for context, the original PR and review comments are in #88

**Review by wcmitchell — APPROVED** [review]
Looks good to me. Agree with @coderbydesign in #88 on eventually breaking out the metrics into a separate....piece? Module? Something, but that's nothing that should hold up this going in.

File: apispec/api.spec.json
Comment (coderbydesign) [inline apispec/api.spec.json:78]: Sort of a nit, but should we name these `sample*` responses more explicitly? Looking at them when validating the spec in a swagger editor seemed like it might be more readable if they were intention revealing.

Comment (coderbydesign) [reply apispec/api.spec.json:78]: Also, should we include the potential for a dependency failure response? https://github.com/RedHatInsights/entitlements-api-go/pull/90/files#diff-e7ac0a2dd3ec503a6608c98c6bee18b71671bcae72450a5fa307bbaa55698297R122

File: config/main.go
Comment (coderbydesign) [inline config/main.go:54]: Should we add anything to the README around testing this in dev/qa/stage with certs and updating your `ENT_COMPLIANCE_HOST` locally?

**Review by coderbydesign — COMMENTED** [review]
A few additional minor comments/questions, but this is looking awesome! I tested with the test users provided from legal and verified the expected error responses in those cases, as well as a bad identity header, and happy path.

File: Dockerfile
Comment (dagbay-rh) [reply Dockerfile:30]: oh, yeah I meant to revisit this. I was having trouble running the image but it was because I wasn't properly importing my keys so it was looking for the default test keys in `test_data`. I can remove this

- [ ] remove import of test data

File: apispec/api.spec.json
Comment (dagbay-rh) [reply apispec/api.spec.json:78]: Yep, good catch. I'll add both of those

- [ ] rename sample responses to reveal intention
- [ ] add dependency failure response to api spec

File: config/main.go
Comment (dagbay-rh) [reply config/main.go:54]: for sure, I can add a blurb about this

- [ ] update readme to include instructions around using compliance config

**Review by coderbydesign — APPROVED** [review]
Looks good @dagbay-rh! Since this will roll out to stage automatically once merged, it's probably worth updating app-interface to set the `ENT_COMPLIANCE_HOST` param in all environments so the code will just pick them up once it's deployed?

**Comment by dagbay-rh** [comment]
> Looks good @dagbay-rh! Since this will roll out to stage automatically once merged, it's probably worth updating app-interface to set the `ENT_COMPLIANCE_HOST` param in all environments so the code will just pick them up once it's deployed?

PR here: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/45816

**Pull request merged**
